### PR TITLE
Fix clusterresourcequota status deepcopy

### PR DESCRIPTION
**Pull request by liggitt**
fixes https://github.com/openshift/origin/issues/11560
based on #11621

### CODE REVIEW

**Comment by liggitt** [comment]
@smarterclayton @deads2k PTAL

**Comment by liggitt** [comment]
[test]

**Comment by smarterclayton** [comment]
LGTM

File: pkg/quota/api/types.go [hunk 1]
```diff
@@ -115,6 +116,26 @@ func (o *ResourceQuotasStatusByNamespace) OrderedKeys() *list.List {
 	return o.orderedMap.OrderedKeys()
 }
 
+func init() {
```
Comment (liggitt) [inline pkg/quota/api/types.go]: now that we were actually copying the internal fields, semantic deepequal choked on comparing different unexported fields. we need to tell it how to compare our custom type. not sure where to hook this (AddKnownTypes? init() here?)

File: pkg/quota/api/zz_generated.deepcopy.go [hunk 2]
```diff
@@ -5,11 +5,12 @@
 package api
 
 import (
+	reflect "reflect"
```
Comment (deads2k) [inline pkg/quota/api/zz_generated.deepcopy.go]: this is what the generator produces now?

**Comment by deads2k** [comment]
just the question about deepcopy generation not conflicting.

lgtm

File: pkg/quota/api/zz_generated.deepcopy.go (hunk 2 above)
Comment (liggitt) [inline pkg/quota/api/zz_generated.deepcopy.go]: fixed up

**Comment by liggitt** [comment]
will pick to 1.3.x/3.3.x once tests are green

**Comment by liggitt** [comment]
[merge]

**Pull request merged**